javatests/arcs/core/entity/HandleManagerTestBase.kt

```diff
@@ -734,10 +734,59 @@ open class HandleManagerTestBase {
 
         val updateDeferred = readHandle.onUpdateDeferred { it.size == 1 }
         writeHandle.dispatchStore(entity)
-        val entityOut = updateDeferred.await()
         assertThat(updateDeferred.await()).containsExactly(entity)
     }
 
+    @Test
+    fun collectionsOfReferencesWorkEndToEnd() = testRunner {
+        fun toReferencedEntity(value: Int) = TestReferencesParticle_Entities_References(value)
+
+        val referencedEntitiesKey = ReferenceModeStorageKey(
+            backingKey = RamDiskStorageKey("referencedEntities"),
+            storageKey = RamDiskStorageKey("set-referencedEntities")
+        )
+
+        val referencedEntityHandle = writeHandleManager.createCollectionHandle(
+            referencedEntitiesKey,
+            entitySpec = TestReferencesParticle_Entities_References
+        )
+
+        suspend fun toReferences(values: Iterable<Int>) = values
+            .map { toReferencedEntity(it) }
+            .map {
+                referencedEntityHandle.dispatchStore(it)
+                referencedEntityHandle.dispatchCreateReference(it)
+            }
+
+        suspend fun toEntity(values: Set<Int>, valueList: List<Int>) =
+            TestReferencesParticle_Entities(
+                toReferences(values).toSet(),
+                toReferences(valueList)
+            )
+
+        val entities = setOf(
+            toEntity(setOf(1, 2, 3), listOf(3, 3, 4)),
+            toEntity(setOf(200, 100, 300), listOf(2, 10, 2)),
+            toEntity(setOf(34, 2145, 1, 11), listOf(3, 4, 5))
+        )
+
+        val writeHandle = writeHandleManager.createCollectionHandle(
+            entitySpec = TestReferencesParticle_Entities
+        )
+        val readHandle = readHandleManager.createCollectionHandle(
+            entitySpec = TestReferencesParticle_Entities
+        )
+
+        val updateDeferred = readHandle.onUpdateDeferred { it.size == 3 }
+        entities.forEach { writeHandle.dispatchStore(it) }
+        val entitiesOut = updateDeferred.await()
+        assertThat(entitiesOut).containsExactly(*entities.toTypedArray())
```

Comment:
does `assertThat(entitiesOut).containsExactlyElementsIn​(entities)` work?

Reply:
it does, thanks.